sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -947,19 +1038,17 @@ private Channel pollChannel() {
             return first;  // because this.close -> this.close0 -> this.pollChannel
         }
 
-        if (this.isChannelServiceable(first)) {
+        if (this.isChannelServiceable(first, false)) {
             return first;
         }
 
         this.availableChannels.offer(first);  // because we need a non-null sentinel to stop the search for a channel
 
-        for (Channel next = this.availableChannels.pollLast(); next != first; next =
-            this.availableChannels.pollLast()) {
-
+        for (Channel next = this.availableChannels.pollLast(); next != first; next = this.availableChannels.pollLast()) {
             assert next != null : "impossible";
 
             if (next.isActive()) {
```

Comment:
Clarification: Any idea on why the first channel short circuit has a different pre-condition (isClosed())?

Reply:
The is a comment in the short-circuit above already - to avoid an infinite loop in the close code sequence.